src/main/java/com/lithium/dbi/rdbi/recipes/scheduler/DedupJobScheduler.java

```diff
@@ -0,0 +1,100 @@
+package com.lithium.dbi.rdbi.recipes.scheduler;
+
+import com.lithium.dbi.rdbi.Handle;
+import com.lithium.dbi.rdbi.RDBI;
+import org.joda.time.Instant;
+
+import java.util.List;
+
+/**
+ * Similar to ExclusiveJobScheduler. Main difference is that jobs are de-duplicates by state.
+ * So the same job might be in the ready queue and the running queue at the same time. However,
+ * only one instance of a given job will ever be found in a particular state. This allows clients
+ * to schedule a job to run immediately after the one currently in progress has been completed.
+ *
+ * When reserving multiple jobs the scheduler will make sure it returns the requested number of jobs as long
+ * as they are available to be reserved in the ready queue. This is an important difference from how the
+ * PriorityBasedJobScheduler and TimeBasedScheduler work.
+ */
+public class DedupJobScheduler extends AbstractDedupJobScheduler {
+
+    /**
+     * @param rdbi the rdbi driver
+     * @param redisPrefixKey the prefix key for the job system. All keys the job system uses will have the prefix redisPrefixKey
+     */
+    public DedupJobScheduler(RDBI rdbi, String redisPrefixKey) {
+        super(rdbi, redisPrefixKey);
+    }
+
+    @Override
+    public boolean schedule(final String tube, final String jobStr, final int runInMillis) {
+
+        try (Handle handle = rdbi.open()) {
+            return 1 == handle.attach(DedupJobSchedulerDAO.class).scheduleJob(
+                    getReadyQueue(tube),
+                    getPaused(tube),
+                    jobStr,
+                    Instant.now().getMillis() + runInMillis);
+        }
+    }
+
+    @Override
+    public List<TimeJobInfo> reserveMulti(final String tube, final long ttrInMillis, final int maxNumberOfJobs) {
+        try (Handle handle = rdbi.open()) {
+            return handle.attach(DedupJobSchedulerDAO.class).reserveJobs(
+                    getReadyQueue(tube),
+                    getRunningQueue(tube),
+                    getPaused(tube),
+                    maxNumberOfJobs,
+                    Instant.now().getMillis(),
+                    Instant.now().getMillis() + ttrInMillis);
+        }
+    }
+
+    @Override
+    public boolean deleteJob(final String tube, String jobStr) {
+        try (Handle handle = rdbi.open()) {
+            return 1 == handle.attach(DedupJobSchedulerDAO.class)
+                              .deleteJob(getReadyQueue(tube), getRunningQueue(tube), jobStr);
+        }
+    }
+
+    public boolean ackJob(final String tube, String jobStr) {
```

Comment:
ackJob replaces the deleteJob that we have used traditionally in order to ack a running job.